src/Chainweb/SPV/CreateProof.hs

```diff
@@ -435,26 +434,25 @@ crumbsOnChain trgHeader srcHeight
 -- Returns 'Nothing' if no such path exists.
 --
 crumbsToChain
-    :: Given WebBlockHeaderDb
-    => ChainId
+    :: WebBlockHeaderDb
+    -> ChainId
     -> BlockHeader
     -> IO (Maybe (BlockHeader, [(Int, BlockHeader)]))
         -- ^ bread crumbs that lead from to source Chain to targetHeader
-crumbsToChain srcCid trgHeader
+crumbsToChain db srcCid trgHeader
     | (int (_blockHeight trgHeader) + 1) < length path = return Nothing
     | otherwise = Just <$> go trgHeader path []
   where
-    graph = _chainGraph @WebBlockHeaderDb given
-    path = shortestPath (_chainId trgHeader) srcCid graph
+    path = shortestPath (_chainId trgHeader) srcCid (_chainGraph db)
 
     go
        :: BlockHeader
        -> [ChainId]
        -> [(Int, BlockHeader)]
        -> IO (BlockHeader, [(Int, BlockHeader)])
-    go !cur [] !acc = return $! (cur, acc)
+    go !cur [] !acc = return (cur, acc)
```

Comment:
we sure we don't want to force this?

Reply:
`(,)` is a constructor. There's nothing to force.